src/Codec/Archive/Zip.hs

```diff
@@ -357,13 +362,27 @@ checkPath fp =
                     (_:ys) -> return ys
           _    -> return (x:xs)
 
--- | Writes contents of an 'Entry' to a file.  Throws a
--- 'CRC32Mismatch' exception if the CRC32 checksum for the entry
--- does not match the uncompressed data.
+-- | Writes contents of an 'Entry' to a file.
+-- Uses either writeRegularEntry to write normal files, or
+-- writeSymbolicLinkEntry to write symbolic links if platform supported and
+-- a symbolic link option is specified.
```

Comment:
Since writeRegularEntry and writeSymbolicLink entry aren't exported, I think it would be better if this comment didn't mention them.  (They are implementation details.)  Instead, it should describe what the function does, both for symbolic links and for regular files.  (So, the content in the old comment, about CRC32Mismatch, should still be here, for example.)  Remember that this is what the library user will read in the API documentation.

Reply:
:+1: I've corrected the docs for the methods here, let me know if it's better now

Reply:
I committed some wording changes to your branch.